iscsi/targetd/provisioner/iscsi-provisioner.go

```diff
@@ -108,6 +108,40 @@ func (p *iscsiProvisioner) getAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
+func createPVSource(options controller.VolumeOptions, portals []string, lun int32) v1.PersistentVolumeSource {
+	if options.PVC.Annotations["os"] == "windows" {
```

Comment:
I think this merits some discussion: 
1. For multi-OS support from a single provisioner, is the recommendation to use an annotation on the PVC? If so maybe the annotation can be standardized with appropriate prefixes. Advantage of this is you get to keep a single storage class and service PVCs from that.
2. An alternative could be to have a storage class per OS and pass the OS a parameter in the storage class that the external provisioner processes.
Thoughts? @jsafrane @jingxu97

Reply:
If the cluster is heterogenous (i.e. Windows and Linux nodes mixed together), I would expect that WaitForFirstConsumer is used and the provisioner creates a PV that can be used from the node where corresponding pod is scheduled (i.e. the provisioner checks Node `kubernetes.io/os` label.

With Immediate mode, the provisioner can use a config option or StorageClass parameter to determine what to provision. Or simply roll a dice. Annotations in PVC make the PVC dependent on the platform and that's not what we want.

In all cases (both Immediate & WaitForFirstConsumer provisioning), the provisioned PV should have node affinity that selects nodes with the right OS (e.g. `kubernetes.io/os: linux`)